File: day2/main.py
with open("t.txt", "r") as f:
    parts = f.read().strip().split(",")


def part2(a, b):
    ans = 0
    for j in range(a, b + 1):
        s = str(j)
        for x in range(1, len(s)):
            sub = s[:x]
            if sub * (len(s) // len(sub)) == s:
                ans += int(s)
    return ans


def part1():
    ans = 0
    total = 0
    for i in parts:
        a, b = i.split("-")
        a = int(a)
        b = int(b)

        total += part2(a, b)

        for j in range(a, b + 1):
            s = str(j)
            if len(s) % 2 != 0:
                continue
            mid = len(s) // 2
            first_half = s[:mid]
            if first_half * 2 == s:
                ans += j

    print(ans)
    print(total)


part1()

File: day2/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func part2(a int, b int) int {
	ans := 0
	for j := a; j <= b; j++ {
		ns := strconv.Itoa(j)
		for x := 1; x < len(ns); x++ {
			sub := ns[:x]
			if sub == "" {
				continue
			}
			repeatTimes := len(ns) / len(sub)

			built := strings.Repeat(sub, repeatTimes)
			if built == ns {
				val, _ := strconv.Atoi(ns)
				ans += val
			}
		}
	}
	return ans
}

func main() {
	file, _ := os.Open("t.txt")
	scanner := bufio.NewScanner(file)

	total := 0
	ans := 0
	for scanner.Scan() {
		line := scanner.Text()
		parts := strings.SplitSeq(line, ",")
		for v := range parts {
			r := strings.Split(v, "-")

			a, _ := strconv.Atoi(r[0])
			b, _ := strconv.Atoi(r[1])
			total += part2(a, b)

			for j := a; j <= b; j++ {
				s := strconv.Itoa(j)
				if len(s)%2 != 0 {
					continue
				}
				mid := len(s) / 2
				news := s[:mid] + s[:mid]

				if news == s {
					ans += j
				}
			}
		}
	}
	fmt.Println(ans)
	fmt.Println(total)
}
